src/components/ui/tooltip.tsx

```diff
@@ -0,0 +1,67 @@
+'use client';
+
+import * as React from 'react';
+import * as TooltipPrimitive from '@radix-ui/react-tooltip';
+import { X } from 'lucide-react';
+
+interface TooltipProps {
+  children: React.ReactNode;
+  tooltipContent: string;
+  tooltipTitle: string;
+}
+
+export function Tooltip({
+  children,
+  tooltipContent,
+  tooltipTitle,
+}: TooltipProps) {
+  const [isOpen, setIsOpen] = React.useState(false);
+  const isTouchDevice = React.useRef(false);
+
+  React.useEffect(() => {
+    isTouchDevice.current =
+      'ontouchstart' in window || navigator.maxTouchPoints > 0;
+  }, []);
```

Comment:
Not a deal breaker but the condition could just be checked by a helper during the onclick instead of ref+effect